cmd/stats-exporter/main.go

```diff
@@ -0,0 +1,114 @@
+package main
+
+import (
+	"database/sql"
+	"flag"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"strings"
+	"time"
+
+	"github.com/letsencrypt/boulder/cmd"
+)
+
+type sqlRows interface {
+	Next() bool
+	Scan(dest ...interface{}) error
+}
+
+type dbQueryable interface {
+	Query(string, ...interface{}) (*sql.Rows, error)
+}
+
+var sqlOpen = func(driver, dsn string) (dbQueryable, error) {
+	return sql.Open(driver, dsn)
+}
+
+var execRun = func(c *exec.Cmd) error {
+	return c.Run()
+}
+
+func queryDB(dbConnect, beginTimeStamp, endTimeStamp string) (*sql.Rows, error) {
+	dbDSN, err := ioutil.ReadFile(dbConnect)
+	if err != nil {
+		return nil, fmt.Errorf("Could not open database connection file: %s", err)
+	}
+	db, err := sqlOpen("mysql", strings.TrimSpace(string(dbDSN)))
+	if err != nil {
+		return nil, fmt.Errorf("Could not establish database connection: %s", err)
+	}
+	rows, err := db.Query(`SELECT id,reversedName,notBefore,serial FROM issuedNames where notBefore >= ? and notBefore < ?`, beginTimeStamp, endTimeStamp)
+	if err != nil {
+		return nil, fmt.Errorf("Could not reach database and run query: %s", err)
+	}
+	return rows, nil
+}
+
+func writeTSVData(rows sqlRows, outFile io.Writer) error {
+	for rows.Next() {
+		var (
+			id, rname, notBefore, serial string
+		)
+		if err := rows.Scan(&id, &rname, &notBefore, &serial); err != nil {
+			return err
+		}
+		if _, err := fmt.Fprintf(outFile, "%s\t%s\t%s\t%s\n", id, rname, notBefore, serial); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func compress(outputFileName string) error {
+	gzipCmd := exec.Command("/usr/bin/gzip", outputFileName)
+	err := execRun(gzipCmd)
+	if err != nil {
+		return fmt.Errorf("Could not gzip result file: %s", err)
+	}
+	return nil
+}
+
+func scp(outputFileName, destination, key string) error {
+	outputGZIPName := outputFileName + ".gz"
+	scpCmd := exec.Command("/usr/bin/scp", "-i", key, outputGZIPName, destination)
```

Comment:
Two suggestions:
1) like for `gzip` I think its more robust to use `scp` instead of hardcoding the path.
2) I don't think its a good idea to bake in an `-i` argument for an identity file. It would be better for the operator to configure their `ssh-agent` or `~/.ssh/config` file to provide the correct identity for connecting to the host in question.  This is more flexible, works across any uses of `scp`/`ssh` and reduces the complexity of this program.
```suggestion
	scpCmd := exec.Command("scp", outputGZIPName, destination)
```

Reply:
That's a great suggestion! I didn't like having to provide the key argument either and didn't think about configuring that with an ssh config. My only concern is that a system user will likely be running this script and would not have a home directory. I'm not sure if a custom ssh-setup would be the right path

Reply:
Ahhh! I see, that's a good concern. I hadn't thought about a user without a home directory. I think with that in mind if the `exec.Command` for `scp` stays in-scope having a `-i` arg piped through makes sense to me :+1: Thanks!